libs/block-content/website/src/lib/streamable/streamable-video-block.tsx

```diff
@@ -0,0 +1,104 @@
+import styled from '@emotion/styled';
+import {
+  BlockContent,
+  StreamableVideoBlock as StreamableVideoBlockType,
+} from '@wepublish/website/api';
+import { BuilderStreamableVideoBlockProps } from '@wepublish/website/builder';
+import React, { useEffect, useMemo, useState } from 'react';
+import ReactPlayer from 'react-player';
+
+export const isStreamableVideoBlock = (
+  block: Pick<BlockContent, '__typename'>
+): block is StreamableVideoBlockType =>
+  block.__typename === 'StreamableVideoBlock';
+
+export const StreamableVideoBlockWrapper = styled('div')``;
+
+const AspectBox = styled('div')<{ $aspectRatio: string }>`
+  width: 100%;
+  /* haelt die Flaeche proportional, egal ob Hoch- oder Querformat */
+  aspect-ratio: ${({ $aspectRatio }) => $aspectRatio || '16/9'};
+
+  ${({ theme }) => theme.breakpoints.between('sm', 'xl')} {
+    max-height: 80vh;
+  }
+
+  /* Inhalt fuellt die Box */
+  & > .react-player {
+    width: 100% !important;
+    height: 100% !important;
+  }
+`;
+
+/**
+ * Ermittelt das Aspect-Ratio-String "w/h".
+ * Faellt zurueck auf 16/9, wenn ungueltige Werte reinkommen.
+ */
+function toAspectRatioString(
+  width?: number,
+  height?: number,
+  fallback = '16/9'
+) {
+  if (!width || !height || width <= 0 || height <= 0) return fallback;
+  return `${Math.round(width)}/${Math.round(height)}`;
+}
+
+export function StreamableVideoBlock({
+  videoID,
+  className,
+}: BuilderStreamableVideoBlockProps) {
+  const [aspectRatio, setAspectRatio] = useState<string>('16/9');
+
+  const streamableUrl = useMemo(() => {
```

Comment:
this is a net loss. A single asignment uses less memory and cpu than using useMemo.